File: CS-21(Java)/Asn 1/Stack.java
/*                                                                                                                                                                                                        
  Subclass of the LinkedList class.                                                                                                                                                                       
  Extends its functionality by adding pop/push methods.                                                                                                                                                   
*/
public class Stack extends LinkedList {

    /*                                                                                                                                                                                                    
      Creates a new Stack.                                                                                                                                                                                
    */
    public Stack() {
        super();
    }

    /*                                                                                                                                                                                                    
      Adds a new double to the beginning of the list.                                                                                                                                                     
    */
    public void push(double d) {
        Node n = new Node(d);
        n.next = head.next;
        head.next = n;
    }

    /*                                                                                                                                                                                                    
      Returns the first Double in the list and removes it.                                                                                                                                                
    */
    public Double pop() {
        Double retval = get(0);
        remove(0);
        return retval;
    }
}

File: CS-21(C++)/Asn 1/Linked List Stack/Stack.cpp
#include "Stack.h"

Stack::Stack() {
	head = 0;
	size = 0;
}

void Stack::push(double d) {
	Node *node = new Node(d);
	
	if (!head) {
		head = node;
	}
	else {
		node->next = head;
		head = node;
	}
	size++;
}

double Stack::pop() {
	Node *temp = head;
	
	double val = head->data;
	head = head->next;
	
	delete temp;
	size--;
	return val;
}

bool Stack::isempty() {
	return (size == 0);
}